### PR TITLE
Utils refactor

### PR DESCRIPTION
**Pull request by kelle**

### CODE REVIEW

**Comment by kelle** [comment]
This is ready for review. I hope we can discuss at co-working tomorrow. :)

File: scripts/ingests/ingest_utils.py [hunk 1]
```diff
@@ -0,0 +1,782 @@
...
+def ingest_sources(db, sources, ras, decs, references, comments=None, epochs=None,
+                   equinoxes=None, raise_error=True):
+    """
+    Script to ingest sources
+
+    Parameters
+    ----------
+    db: astrodbkit2.astrodb.Database
+        Database object created by astrodbkit2
+    sources: list of strings
```
Comment (Will-Cooper) [inline scripts/ingests/ingest_utils.py:27]: Love the type hinting! However, think the syntax is generally: 
```suggestion
    sources: list[str]
```
for example

File: scripts/ingests/ingest_utils.py [hunk 2]
```diff
@@ -0,0 +1,782 @@
...
+            else:
+                msg = f"{i}: Skipping: {source}. Not sure why."
+                msg2 = f"\n {str(source_data)} "
+                logger.warning(msg)
+                logger.debug(msg2)
+                n_skipped += 1
+                if raise_error:
+                    raise SimpleError(msg+msg2)
+                else:
+                    continue
```
Comment (Will-Cooper) [inline scripts/ingests/ingest_utils.py:182]:
```suggestion
                continue
```
technically don't need the else: continue, as the continue will always activate if the raise error is False. Doesn't matter though!

File: scripts/ingests/ingest_utils.py [hunk 3]
```diff
@@ -0,0 +1,782 @@
...
+                    db.Names.insert().execute(alt_names_data)
+                    logger.debug(f"{i}: Name added to database: {alt_names_data}\n")
+                    n_alt_names += 1
+                except sqlalchemy.exc.IntegrityError:
+                    msg = f"{i}: Could not add {alt_names_data} to database"
+                    logger.warning(msg)
+                    if raise_error:
+                        raise SimpleError(msg)
+                    else:
+                        continue
```
Comment (Will-Cooper) [inline scripts/ingests/ingest_utils.py:95]:
```suggestion
                except sqlalchemy.exc.IntegrityError as e:
                    msg = f"{i}: Could not add {alt_names_data} to database"
                    logger.warning(msg)
                    if raise_error:
                        raise SimpleError(msg + '\n' + e)
                    else:
                        continue
```

File: scripts/ingests/ingest_utils.py [hunk 4]
```diff
@@ -0,0 +1,782 @@
...
+        reference = [reference] * len(sources)
+
+    if isinstance(telescope, str):
+        telescope = [telescope] * len(sources)
+
+    if isinstance(instrument, str):
+        instrument = [instrument] * len(sources)
+
+    if isinstance(ucds, str):
+        ucds = [ucds] * len(sources)
```
Comment (Will-Cooper) [inline scripts/ingests/ingest_utils.py:605]: if you like, the following also works
```suggestion
things = [bands, reference, telescope, instrument, ucds]
for i, thing in enumerate(things):
    if isinstance(thing, str):
        thing = [thing] * len(sources)
        things[i] = thing
bands, reference, telescope, instrument, ucds = things
```

File: scripts/ingests/ingest_utils.py [hunk 5]
```diff
@@ -0,0 +1,782 @@
...
+"""
+from scripts.ingests.utils import *
+import logging
+import numpy as np
+import numpy.ma as ma
+from astropy.table import Table, unique
+import re
+import dateutil
+import pandas as pd
+from sqlalchemy import func
```
Comment (Will-Cooper) [inline scripts/ingests/ingest_utils.py:12]:
```suggestion
from astropy.table import Table, unique
import numpy as np
import numpy.ma as ma
import pandas as pd
from sqlalchemy import func

import dateutil
import logging
import re

from scripts.ingests.utils import *
```

File: tests/test_utils.py [hunk 6]
```diff
@@ -36,9 +37,9 @@ def db():
 # Create fake astropy Table of data to load
 @pytest.fixture(scope="module")
 def t_plx():
-    t_plx = Table([{'source': 'Fake 1', 'plx': 113, 'plx_err': 0.3, 'plx_ref': 'Ref 1'},
-                   {'source': 'Fake 2', 'plx': 145, 'plx_err': 0.5, 'plx_ref': 'Ref 1'},
-                   {'source': 'Fake 3', 'plx': 155, 'plx_err': 0.6, 'plx_ref': 'Ref 2'},
+    t_plx = Table([{'source': 'Fake 1', 'plx': 113., 'plx_err': 0.3, 'plx_ref': 'Ref 1'},
+                   {'source': 'Fake 2', 'plx': 145., 'plx_err': 0.5, 'plx_ref': 'Ref 1'},
+                   {'source': 'Fake 3', 'plx': 155., 'plx_err': 0.6, 'plx_ref': 'Ref 2'},
```
Comment (Will-Cooper) [inline tests/test_utils.py:42]: why the float casting?

**Review by Will-Cooper — COMMENTED** [review]
Looking great!

File: scripts/ingests/ingest_utils.py (hunk 2 above)
Comment (kelle) [reply scripts/ingests/ingest_utils.py:184]: I'm gonna leave it. :)

File: tests/test_utils.py (hunk 6 above)
Comment (kelle) [reply tests/test_utils.py:42]: Integer parallaxes are unlikely.

**Comment by kelle** [comment]
@Will-Cooper , I implemented a couple of your suggestions in more places. Please re-review and/or re-approve so I can merge. Thanks!

**Review by Will-Cooper — APPROVED** [review]
LGTM!
One final little thing with the type hinting is you can do instead of:
```python
str or list[str]
```
put
```python
from typing import Union
Union[str, list[str]]
```

**Pull request merged**